### PR TITLE
[CoreBluetooth] Update bindings for Xcode 11 beta 1 & beta 2.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Updated the bindings to match Xcode 11 and took the time to add a StrongDict (new keys were added) to improve the API a little.

### CODE REVIEW

File: src/corebluetooth.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 //
 // Copyright 2011-2013 Xamarin Inc
 //
+using System.ComponentModel;
+
```
Comment (whitneyschmidt) [inline src/corebluetooth.cs]: extra line intended?

File: src/corebluetooth.cs [hunk 2]
```diff
@@ -7,6 +7,8 @@
 //
 // Copyright 2011-2013 Xamarin Inc
```
Comment (whitneyschmidt) [inline src/corebluetooth.cs]: nit: update

File: src/corebluetooth.cs [hunk 3]
```diff
@@ -43,6 +45,59 @@ interface CBCentralInitOptions {
 	interface CBManager {
 		[Export ("state", ArgumentSemantic.Assign)]
 		CBManagerState State { get; }
+
+		[iOS (13,0), TV (13,0), Watch (6,0), Mac (10, 15)]
```
Comment (spouliot) [inline src/corebluetooth.cs]: I think we still need the `only64: true` on the bindings...

File: src/corebluetooth.cs [hunk 4]
```diff
@@ -43,6 +45,59 @@ interface CBCentralInitOptions {
...
+		[Export ("authorization", ArgumentSemantic.Assign)]
+		CBManagerAuthorization Authorization { get; }
+	}
+
+	[iOS (13,0), TV (13,0), Watch (6,0)]
```
Comment (spouliot) [inline src/corebluetooth.cs]: missing `[NoMac]` ? unless it's present (then missing `Mac (10,15, onlyOn64: true)]` or you're inside `!MONOMAC` (then better add `[NoMac]` anyway to ease review and help the IDE)

File: src/corebluetooth.cs [hunk 5]
```diff
@@ -43,6 +45,59 @@ interface CBCentralInitOptions {
...
+	[StrongDictionary ("CBConnectionEventMatchingOptionsKeys")]
+	interface CBConnectionEventMatchingOptions {
+		NSUuid [] PeripheralUuids { get; set; }
+		CBUUID [] ServiceUuids { get; set;}
+	}
+
+	[iOS (13,0), TV (13,0), Watch (6,0)]
```
Comment (spouliot) [inline src/corebluetooth.cs]: same

File: src/corebluetooth.cs [hunk 6]
```diff
@@ -43,6 +45,59 @@ interface CBCentralInitOptions {
...
+		NSString PeripheralUuidsKey { get; }
+		[Field ("CBConnectionEventMatchingOptionServiceUUIDs")]
+		NSString ServiceUuidsKey { get; }
+	}
+
+	[StrongDictionary ("CBConnectPeripheralOptionsKeys")]
+	interface CBConnectPeripheralOptions {
+		[Mac (10,13)] [iOS (6,0)]
+		bool NotifyOnConnection { get; set; }
+		bool NotifyOnDisconnection { get; set; }
```
Comment (spouliot) [inline src/corebluetooth.cs]: that one has no availability ? looks weird (add comment if that's the case)
if possible move some (all) availability attributes on the type itself

File: src/corebluetooth.cs [hunk 7]
```diff
@@ -43,6 +45,59 @@ interface CBCentralInitOptions {
...
+	}
+
+	[Static]
+	[Internal]
+	interface CBConnectPeripheralOptionsKeys {
+		[Mac (10,13)] [iOS (6,0)]
+		[Field ("CBConnectPeripheralOptionNotifyOnConnectionKey")]
+		NSString NotifyOnConnectionKey { get; }
+		[Field ("CBConnectPeripheralOptionNotifyOnDisconnectionKey")]
+		NSString NotifyOnDisconnectionKey { get; }
```
Comment (spouliot) [inline src/corebluetooth.cs]: same

File: src/corebluetooth.cs [hunk 8]
```diff
@@ -858,6 +946,8 @@ interface CBPeripheralManager {
 		NSString RestoredStateAdvertisementDataKey { get; }
 
 #if !MONOMAC || !XAMCORE_4_0
+		[Deprecated (PlatformName.iOS, 13, 0, message: "Use 'CBManager.Authorization' instead.")]
+		[Deprecated (PlatformName.MacOSX, 10, 15, message: "Use 'CBManager.Authorization' instead.")]
```
Comment (spouliot) [inline src/corebluetooth.cs]: looks like tvOS should also have the deprecation

**Comment by spouliot** [comment]
should be update tp beta 3 before merging

File: src/corebluetooth.cs (hunk 3 above)
Comment (rolfbjarne) [reply src/corebluetooth.cs]: Actually if the build doesn't fail and tests don't fail, it shouldn't matter anymore (and any lingering need will be resolved before the final release anyway; it's on our todo list).

Comment (mandel-macaque) [reply src/corebluetooth.cs]: I suppose since that we are not longer supporting 32b, it makes sense, nevertheless, I'll add it. It won't hurt

**Comment by mandel-macaque** [comment]
@spouliot I checked the headers, looks like it beta3 ready since they just changed the availability macro. :) :success:

**Comment by spouliot** [comment]
@mandel-macaque nice, please update the wiki and the description (when you merge)

File: src/CoreBluetooth/Enums.cs [hunk 9]
```diff
@@ -172,4 +172,26 @@ public enum CBPeripheralManagerConnectionLatency : long {
...
 	}
+
+	[iOS (13,0), TV (13,0), Watch (6,0), Mac (10,15)]
+	[Native]
+	public enum CBConnectionEvent : long {
+		Disconnected = 0,
+		Connected = 1,
+	}
+
+	[Flags, iOS (13,0), TV (13,0), Watch (6,0)]
```
Comment (spouliot) [inline src/CoreBluetooth/Enums.cs]: Mac (No or version) is unspecified

File: src/corebluetooth.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/corebluetooth.cs]: yes, I like to separate the bcl using from ours

File: src/corebluetooth.cs (hunk 6 above)
Comment (mandel-macaque) [reply src/corebluetooth.cs]: That is the only one that had no availability attributes from the old constant used: https://github.com/xamarin/xamarin-macios/blob/master/src/corebluetooth.cs#L105 

The other attributes I used from the already present (set to obsolete) constants. So I don't think I can move [Mac (10,13)] [iOS (6,0)] to the top of the class :/

**Comment by mandel-macaque** [comment]
@spouliot my fault, I though that [Introduced (PlatformName.UIKitForMac, 13,0)] would be the correct thing to do.

**Comment by mandel-macaque** [comment]
Midi test failures are un related, I have seen this in other PRs. Looks like a device doing something funny.

**Comment by spouliot** [comment]
It's `[NoMac]` not `[NotMac]`
Please build and test locally before committing :)

**Comment by spouliot** [comment]
@mandel-macaque it might be `[Introduced (PlatformName.UIKitForMac, 13,0)]` even if not available on macOS itself (a bit weird for CoreBluetooth)

**Pull request merged**